src/com/google/enterprise/adaptor/ContentTransformFactory.java

```diff
@@ -0,0 +1,78 @@
+package com.google.enterprise.adaptor;
+
+import com.google.common.base.Strings;
+
+import java.io.OutputStream;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * The content transform factory holds all document content transformers
+ * and puts them in series connection.
+ *
+ * @author Dominik Weidenfeld (dominik.weidenfeld@twt.de)
+ */
+public class ContentTransformFactory {
+
+  private List<DocumentContentTransform> transforms;
+
+  public ContentTransformFactory(
+      final List<Map<String, String>> transforms) {
+    if (transforms.size() <= 0) {
+      return;
+    }
+    this.transforms = new LinkedList<DocumentContentTransform>();
+    for (int i = (transforms.size() - 1); i >= 0; i--) {
+      final Map<String, String> tConfig = transforms.get(i);
+      final String className = tConfig.get("class");
+      if (Strings.isNullOrEmpty(className)) {
+        throw new RuntimeException(
+            "Document Content Transform class is missing " + className);
+      }
+      try {
+        //noinspection unchecked
+        final Class<DocumentContentTransform> clazz =
+            (Class<DocumentContentTransform>) Class.forName(className);
+        final Constructor<DocumentContentTransform> constructor =
+            clazz.getConstructor(Map.class);
+        this.transforms.add(constructor.newInstance(tConfig));
```

Comment:
This change is great, but newInstance should be in createPipeline.

The various threads calling getDocContent cannot be using the
same instances of DocumentContentTransform.

Thank you